zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -162,6 +165,17 @@ final class MixedAnalyzingCompiler(
  * of cross Java-Scala compilation.
  */
 object MixedAnalyzingCompiler {
+  private type JarMetadata = (FileTime, Long)
+  // Using paths instead of files as key because they have more lightweight memory consumption
+  private[this] val cacheMetadataJar = new ConcurrentHashMap[File, (JarMetadata, FileHash)]()
```

Comment:
Doesn't the code do something different from what the comment says? The key is a `File`.

Reply:
and that is why comments SUCK

Reply:
Yup, that's a stale comment, thanks for the catch.